proxy/contracts/schain/MessageProxyForSchain.sol

```diff
@@ -339,6 +346,32 @@ contract MessageProxyForSchain is MessageProxy, IMessageProxyForSchainInitialize
         return _registryContracts;
     }
 
+    /**
+     * @dev Returns Etherbase contract
+     */
+    function _getEtherbase() internal view virtual returns (IEtherbaseUpgradeable) {
+        return ETHERBASE;
+    }
+
+    /**
+     * @dev Move skETH from Etherbase if the balance is too low
+     */
+    function _topUpBalance() private {
+        uint balance = msg.sender.balance + gasleft() * tx.gasprice;
```

Comment:
Is balance changed before tx? Why there is an addition instead of subtraction?

Reply:
Before transaction execution a balance is reduced by `gas limit` * `gas price`.
After execution all unspent ETH returns back.

Reply:
Got it, thanks!

Reply:
Why did you add `gasleft() * tx.gasprice` to the balance? As I understand final balance would be a little bit less. So missingAmount would be a little bit less, and a balance after topup would be less than 1 eth.

Reply:
Yes you are right but the difference is small and there is no easy way to get accurate number.
Therefore I did it in this way in order to not overcomplicate. The approach solves the problem. It's not critical to get exactly `MINIMUM_BALANCE` at the end.

Reply:
By the way I'm not sure if `MINIMUM_BALANCE` value is big enough. What do you think?

Reply:
Got it, I think to get balance without addition would be simpler than balance with addition. And the final result would be a little bit more than `MINIMUM_BALANCE` all the time. I think 1 eth is completely enough, minimum gas price on schain is 100 000 Wei which is 0.0001 Gwei is a regular tx gas price for agent. Gas limit is about 250M of gas so max tx size with regular tx gas price would cost 250 000 000 * 0.0001 Gwei = 25 000 Gwei = 0.000025 eth

Reply:
I don't think it's a good idea because both gas limit and gas price is set by an agent. It gives a single agent an opportunity to attack skale chain and to seize all skETH.
Assume a node balance is 10 skETH. During IMA transfer the node sends a transaction with 0.0001 gwei gas price and 10^13 gas limit. In fact it spends a small amount of skETH but during a smart contract execution `balance` is equal to `0` and IMA pulls 1 skETH from `Etherbase` and the node balance becomes ~11 skETH.
Maybe because a `MINIMUM_BALANCE` is pretty small it's not a big issue but by not counting of `gasleft() * tx.gasprice` we open an attack vector and improve nothing.

Reply:
Thanks for explanation - totally agree, did not consider this solution from the attacker perspective